launch_ros/launch_ros/actions/node.py

```diff
@@ -434,6 +442,25 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
                 cmd_extension.extend(['-r', f'{src}:={dst}'])
             self.cmd.extend([normalize_to_list_of_substitutions(x) for x in cmd_extension])
 
+    def _setup_security(
+        self, context: LaunchContext, ros_specific_arguments: Dict[str, Union[str, List[str]]]
+    ):
+        """Enable encryption, creating a key for the node if necessary."""
+        nodl_node = nodl.get_node_by_executable(
+            package_name=self.__package, executable_name=self.__node_executable
+        )
+
+        self.__enclave = self.node_name.replace(
+            Node.UNSPECIFIED_NODE_NAME, nodl_node.name
+        ).replace(Node.UNSPECIFIED_NODE_NAMESPACE, '')
+
+        sros2.keystore.create_enclave(
+            keystore_path=pathlib.Path(os.environ['ROS_SECURITY_KEYSTORE']),
```

Comment:
I thought we were no longer going to rely on these environment variables?

Reply:
These variables are set by the "SetEnvironmentVariable" action within the subshell. They're not from the user's shell.

Reply:
Is it possible to move to using these environment variables to the very edge of integration between launch_ros and when the node is actually launched? Right now we're in a middle place where we don't rely on the user-set variables anymore, but now we're using it as some sort of internal interface. It feels like it would be better to use the context for this type of thing.

Reply:
c40721f